mlir/include/mlir/IR/Builders.h

```diff
@@ -341,6 +342,19 @@ class OpBuilder : public Builder {
     InsertPoint(Block *insertBlock, Block::iterator insertPt)
         : block(insertBlock), point(insertPt) {}
 
+    /// Compute an insertion point to a place that post-dominates the
+    /// definitions of all given values. Returns an "empty" insertion point if
+    /// no such insertion point exists.
+    ///
+    /// There may be multiple suitable insertion points. This function chooses
+    /// an insertion right after one of the given values.
+    ///
+    /// Note: Some of the given values may already have gone out of scope at the
```

Comment:
Note: if there exists an insertion point that post-dominates all definitions and where all values are in scope, this function is guaranteed to find it.